tools/config/__init__.py

```diff
@@ -653,13 +700,13 @@ def regions(self):
         if  ((self.target.bootloader_img or self.target.restrict_size) and
              (self.target.mbed_app_start or self.target.mbed_app_size)):
             raise ConfigException(
-                "target.bootloader_img and target.restirct_size are "
+                "target.bootloader_img and target.restrict_size are "
                 "incompatible with target.mbed_app_start and "
                 "target.mbed_app_size")
         if self.target.bootloader_img or self.target.restrict_size:
-            return self._generate_bootloader_build(*self.rom)
+            return self._generate_bootloader_build(*self._get_all_rom_regions())
```

Comment:
I'm a little surprised by this change. What does it do? Why is it needed?

Reply:
It expects rom_start, rom_size as the output and that is then used. We got rid of rom property, so thats why this call to get those parameters.